### PR TITLE
Cli docs

### PR DESCRIPTION
**Pull request by gauntface**
R: @jeffposnick @addyosmani @jpmedley 

This is necessarily done, but I'm hoping we can land some content and add / re-organise in future PR's.

Let me know if you think this is a stupid idea.

### CODE REVIEW

File: docs/getting-started/index.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+Once you have Node, you can get `sw-cli` from NPM by running:
+
+```
+npm install -g sw-cli
+```
+
+## Generating a Service Worker
+
+To generate a service worker for your project just run
+the CLI with the 'generate:sw' command.
```
Comment (jpmedley) [inline docs/getting-started/index.md]: Do I run it from the root of my project, or somewhere else?

File: docs/getting-started/index.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+### Generating SW in Build Process
+
+If you have a build process, you might want to use the
+`sw-build` module instead of using `sw-cli`.
+
+It allows you to generate the service worker  
+programmatically.
+
+[Learn More About sw-build Here](.)
```
Comment (jpmedley) [inline docs/getting-started/index.md]: @jeffposnick You wrote some build instructions, did you? If the need conversion from Docs, send them over to me.

File: docs/getting-started/index.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+Instead of generating a service worker, you may want some
+of the features provided by `sw-cli` but added to your
+own service worker.
+
+For this you have a few options.
+
+Both `sw-cli` and `sw-build` can be used to produce a
+list of assets in your web app with the revision
+information which can be used for precaching. We have
+[recipes on how to do this](.).
```
Comment (jpmedley) [inline docs/getting-started/index.md]: Borken link again?

File: docs/getting-started/index.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+You can then use `sw-lib` to perform the precaching and
+you can define custom routes and add anything else you
+desire to your service worker.
+
+[Learn More About sw-lib Here](.)
```
Comment (jpmedley) [inline docs/getting-started/index.md]: And another.

File: docs/getting-started/index.md (hunk 1 above)
Comment (gauntface) [inline docs/getting-started/index.md]: root of your project.

File: docs/getting-started/index.md (hunk 2 above)
Comment (gauntface) [inline docs/getting-started/index.md]: I will link up Jeff's docs on here once the PR is landed.

File: docs/getting-started/index.md (hunk 3 above)
Comment (gauntface) [inline docs/getting-started/index.md]: Non existent link as waiting on PR.

File: docs/getting-started/index.md (hunk 4 above)
Comment (gauntface) [inline docs/getting-started/index.md]: Non existent link as waiting on PR for your content..

File: docs/getting-started/index.md (hunk 2 above)
Comment (jeffposnick) [inline docs/getting-started/index.md]: I don't think think it's been published to `gh-pages` yet, but it's landed at https://github.com/GoogleChrome/sw-helpers/blob/master/docs/recipes/build/rollup.md

File: docs/getting-started/index.md (hunk 3 above)
Comment (jeffposnick) [inline docs/getting-started/index.md]: Ditto re: https://github.com/GoogleChrome/sw-helpers/blob/master/docs/recipes/build/rollup.md

**Comment by gauntface** [comment]
@jeffposnick @jpmedley 

Fixed to valid links for everything - can change them when we get more relevant content.

I also added an npm run script "npm run serve" to serve the doc site locally - the recipes needed some front matter to be included in the nav - although I'm not 100% confused that travis is building master branch but will see when this lands.

**Pull request merged**